test/cbmc/proofs/DHCP/DHCPProcessEndPoint/DHCPProcessEndPoint_harness.c

```diff
@@ -41,62 +41,97 @@
 #include "FreeRTOS_UDP_IP.h"
 #include "FreeRTOS_DHCP.h"
 #include "FreeRTOS_ARP.h"
+#include "FreeRTOS_Routing.h"
 
 /* Static members defined in FreeRTOS_DHCP.c */
 extern DHCPData_t xDHCPData;
 extern Socket_t xDHCPSocket;
-void prvCreateDHCPSocket();
+void prvCreateDHCPSocket( NetworkEndPoint_t * pxEndPoint );
+
+/* Signature of function under test (vDHCPProcessEndPoint).
+ * This function is defined as static and thus needs to be
+ * taken out of the file using a CBMC flag. Using that flag
+ * results in mangling of the name.  */
+void __CPROVER_file_local_FreeRTOS_DHCP_c_vDHCPProcessEndPoint( BaseType_t xReset,
+                                                                BaseType_t xDoCheck,
+                                                                NetworkEndPoint_t * pxEndPoint );
 
 /* Static member defined in freertos_api.c */
 #ifdef CBMC_GETNETWORKBUFFER_FAILURE_BOUND
     extern uint32_t GetNetworkBuffer_failure_count;
 #endif
 
 /****************************************************************
-* The signature of the function under test.
+* Abstract vSocketBind function to always return a 0 (meaning pass
+* in Berkeley sockets).
+* The function under test asserts the return to be 0 further down
+* the call graph.
 ****************************************************************/
+BaseType_t vSocketBind( FreeRTOS_Socket_t * pxSocket,
+                        struct freertos_sockaddr * pxBindAddress,
+                        size_t uxAddressLength,
+                        BaseType_t xInternal )
+{
+    return ( BaseType_t ) 0;
+}
 
-void vDHCPProcess( BaseType_t xReset );
 
 /****************************************************************
-* Abstract prvProcessDHCPReplies proved memory safe in ProcessDHCPReplies.
+* Abstract FreeRTOS_socket call to allocate a socket sized chunk.
+* The socket cannot be NULL when DHCP is in process and is asserted
+* by the function under test.
 ****************************************************************/
-
-BaseType_t prvProcessDHCPReplies( BaseType_t xExpectedMessageType )
+Socket_t FreeRTOS_socket( BaseType_t xDomain,
+                          BaseType_t xType,
+                          BaseType_t xProtocol )
 {
-    return nondet_BaseType();
+    void * ptr = malloc( sizeof( Socket_t ) );
+
+    __CPROVER_assume( ptr != NULL );
+
+    return ptr;
 }
 
 /****************************************************************
-* The proof of vDHCPProcess
+* Abstract prvProcessDHCPReplies proved memory safe in ProcessDHCPReplies.
 ****************************************************************/
+BaseType_t prvProcessDHCPReplies( BaseType_t xExpectedMessageType,
```

Comment:
I wonder if there is some tag (something we could grep for) we could use in proof harnesses to mark proofs that do this abstraction with the "proved memory safe" justification.  We are "real soon now" going to have the ability to writing "function contracts" to do this kind of abstraction in a sound way.  It would be nice to come back and "patch up" these proofs.

Reply:
I added "Function Abstraction" as the tag.